legend-engine-application-query/src/main/java/org/finos/legend/engine/application/query/model/DataCubeQuery.java

```diff
@@ -0,0 +1,38 @@
+// Copyright 2020 Goldman Sachs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package org.finos.legend.engine.application.query.model;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+
+import java.util.Map;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class DataCubeQuery
+{
+    public String id;
+    public String name;
+    public String description;
+
+    public Map<String, ?> query;
+    public Map<String, ?> source;
+    public Map<String, ?> executionContext;
+
+    public Long lastUpdatedAt;
+    public Long createdAt;
+    public Long lastOpenAt;
+
+    // We make it clear that we only allow a single owner
+    public String owner;
```

Comment:
could we make this a class structure like activator owner, in case we want to improve this. 
we may want to do another ownership type than just a single one.

Reply:
we could, but that's not the bit I want to focus on in this phase. Doing it as `FunctionActivator.Ownership` will entail many things such as coming up with a framework to make filter like `isMyQuery` more complicated to implement (consider case when we have deployment ID and the user comes to us with a kerberos)

Perhaps, we don't even do ownership at all?